spec/filters/cidr_spec.rb

```diff
@@ -192,4 +192,72 @@
       )
     end
   end
+
+  describe "address_field config option" do
+
+    context 'the field is top-level' do
+
+      config <<-CONFIG
+        filter {
+          cidr {
+            address_field => "ip"
+            network => [ "192.168.0.0/24" ]
+            add_tag => [ "matched" ]
+          }
+        }
+      CONFIG
+
+      context 'the input value is a string' do
+        sample({ "ip" => "192.168.0.1" }) do
+          insist { subject.get("tags") }.include?("matched")
+        end
+      end
+
+      context 'the input value is an array' do
+        sample({ "ip" => [ "188.168.0.1", "192.168.0.1" ] }) do
+          insist { subject.get("tags") }.include?("matched")
+        end
+      end
+
+      context 'the input value contains an invalid ip' do
+        sample({ "ip" => [ "invalid", "192.168.0.1" ] }) do
+          insist { subject.get("tags") }.include?("matched")
+        end
+      end
+    end
```

Comment:
I'm a little rusty on [the syntax for negative assertions](https://www.rubydoc.info/gems/insist/1.0.0/Reject) (the insist gem is not used in many plugins), but it would be good to have some edge-case coverage for values that are either missing or present but of unacceptable shape.

```suggestion
      context 'the input value is not present on the event' do
        sample({}) do
          reject { subject.get("tags") }.include?("matched")
        end
      end

      context 'the input value is of unacceptable shape (integer)' do
        sample({ "ip" => 377 }) do
          reject { subject.get("tags") }.include?("matched")
        end
      end
      
      context 'the input value is of unacceptable shape (map)' do
        sample({ "ip" => { "invalid" => "192.168.0.1" }) do
          reject { subject.get("tags") }.include?("matched")
        end
      end
    end
```